lib/mayaUsd/nodes/stageNode.cpp

```diff
@@ -109,7 +109,10 @@ MStatus UsdMayaStageNode::compute(const MPlug& plug, MDataBlock& dataBlock)
             UsdStageCacheContext ctx(UsdMayaStageCache::Get(loadAll));
             usdStage = UsdStage::Open(rootLayer, ArGetResolver().GetCurrentContext());
```

Comment:
I would think similar to what's in proxyShapeBase.cpp, we should probably be checking permissions and possibly creating the `sessionLayer` first, and then use the `Open()` overload that takes a sessionLayer if we created one. Otherwise the anonymous layer created below is not part of the stage's layer stack.

Reply:
Makes sense.